Libraries/CommonFunctions.psm1

```diff
@@ -138,9 +138,9 @@ Function Collect-TestCases($TestXMLs, $TestCategory, $TestArea, $TestNames, $Tes
             if ($TestPlatform -eq "Any") {
                 $platformMatched = $true
             } else {
-                $platforms = $test.Platform.Split(",")
+                $platforms = $test.Platform.Trim(', ').Split(',').Trim()
                 for ($i=0; $i -lt $platforms.length; $i++) {
-                    if ($TestPlatform.Contains($platforms[$i]) -or $platforms[$i].Contains($TestPlatform)) {
+                    if (($TestPlatform -imatch $platforms[$i]) -or ($platforms[$i] -imatch $TestPlatform)) {
```

Comment:
Changing .Contains method to -imatch will create problems if you give, say AzureStack platform bug supported platforms are Azure, AzureStackHub. Now, because, AzureStack "imatches" with AzureStackHub, it will continue as supported platform. That is why, "Contains" method was used to match entire word.

Reply:
I was thinking the same case actually. "AzureStack" -imatch "AzureStackHub" returns $false, which seems to be fine.
@harz566  can you please do more testing to double check the scenario here?

Reply:
@iamshital I already tested that existing, actually previously logic is not elegant from my opinion. To support case insensitive way.  Thinking about your provided example, because there was '-or' in the middle previously, so it always return true previously.

Reply:
See this - 
```
PS C:\Users\shsaveka> $SupportedPlatforms= @("Azure","HyperV")
PS C:\Users\shsaveka> $Platform = "AzureStack"
PS C:\Users\shsaveka> $Platform -imatch $SupportedPlatforms[0]
True <<< (false detection)
PS C:\Users\shsaveka> $Platform -imatch $SupportedPlatforms[1]
False
```
Now you can fix this by reversing the order of -imatch
```
PS C:\Users\shsaveka> $SupportedPlatforms[0] -imatch $Platform
False
```
But again, this is because of -imatch uses substring matching. And this is dangerous.
```
PS C:\Users\shsaveka> $SupportedPlatforms.Contains($Platform)
False
```
Originally, for platform matching only this code was getting used, without for loop.
```
if (!($test.Platform.Split(",").Contains($TestPlatform))) {
    continue
}
```
If you just want to make it case insensitive, then just convert them upper or lower before match - 
```
PS C:\Users\shsaveka> $Platform = "AZURE"
PS C:\Users\shsaveka> $SupportedPlatforms.Contains($Platform)
False
PS C:\Users\shsaveka> $SupportedPlatforms.ToLower().Contains($Platform.ToLower())
True
```
Anyways, I'm just trying to avoid any future problems, if we add platform which has substring of supported platform. :-)

Reply:
Same goes for testcase name matching as well.